sdk/servicebus/service-bus/src/util/utils.ts

```diff
@@ -180,3 +182,214 @@ export function getAssociatedReceiverName(
+
+/**
+ * Represents type of message count details in ATOM based management operations
+ */
+export type MessageCountDetails = {
+  activeMessageCount: number;
+  deadLetterMessageCount: number;
+  scheduledMessageCount: number;
+  transferMessageCount: number;
+  transferDeadLetterMessageCount: number;
+};
+
+/**
+ * Represents type of `AuthorizationRule` in ATOM based management operations
+ */
+export type AuthorizationRule = {
+  claimType: string;
+  claimValue: string;
+  rights: { accessRights?: string[] };
+  keyName: string;
+  primaryKey?: string;
+  secondaryKey?: string;
+};
+
+/**
+ *  @ignore
+ * Helper utility to retrieve array of `AuthorizationRule` from given input,
+ * or undefined if not passed in.
+ * @param value
+ */
+export function getAuthorizationRulesOrUndefined(value: any): AuthorizationRule[] | undefined {
+  const authorizationRules: AuthorizationRule[] = [];
+  const jsonValue: any = getJSONOrUndefined(value);
+  if (jsonValue == undefined) {
+    return undefined;
+  } else {
+    try {
+      const rawAuthorizationRules = value["AuthorizationRule"];
+      if (rawAuthorizationRules.length !== undefined) {
+        for (let i = 0; i < rawAuthorizationRules.length; i++) {
+          authorizationRules.push(buildAuthorizationRule(rawAuthorizationRules[i]));
+        }
+      } else {
+        authorizationRules.push(buildAuthorizationRule(rawAuthorizationRules));
+      }
+      return authorizationRules;
+    } catch (err) {
+      throw new TypeError(
+        `${jsonValue} expected to be in expected to be an array of AuthorizationRule instances, or undefined :: ${err.message}`
+      );
+    }
+  }
+}
+
+/**
+ * Helper utility to build an instance of parsed authorization rule as `AuthorizationRule` from given input,
+ * @param value
+ */
+function buildAuthorizationRule(value: any): AuthorizationRule {
+  const authorizationRule: AuthorizationRule = {
+    claimType: value["ClaimType"],
+    claimValue: value["ClaimValue"],
+    rights: {
+      accessRights: value["Rights"]["AccessRights"]
+    },
+    keyName: value["KeyName"],
+    primaryKey: value["PrimaryKey"],
+    secondaryKey: value["SecondaryKey"]
+  };
+  return authorizationRule;
+}
+
+/**
+ *  @ignore
+ * Helper utility to extract output containing array of `RawAuthorizationRule` instances from given input,
+ * or undefined if not passed in.
+ * @param value
+ */
+export function getRawAuthorizationRules(authorizationRules: AuthorizationRule[] | undefined): any {
+  if (
+    authorizationRules == undefined ||
+    (authorizationRules && authorizationRules.length && authorizationRules.length == 0)
+  ) {
+    return undefined;
+  }
+  const rawAuthorizationRules: any[] = [];
+  if (authorizationRules.length == 1) {
+    rawAuthorizationRules.push(buildRawAuthorizationRule(authorizationRules[0]));
+  } else {
+    for (let i = 0; i < authorizationRules.length; i++) {
+      rawAuthorizationRules.push(buildRawAuthorizationRule(authorizationRules[i]));
```

Comment:
Should we perhaps skip cases when `authorizationRules[i]` is not json object like undefined/null/string/number/boolean?

Reply:
The input is always an `AuthorizationRule`, so the internal utility steps shouldn't require it.

Reply:
This is addressed with updates to getJSObjectOrUndefined helper.
The service accepts empty string as valid entry